services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/permissions/HankeKayttajaService.kt

```diff
@@ -59,7 +85,12 @@ class HankeKayttajaService(
                 .flatMap { it.alikontaktit }
                 .mapNotNull { userContactOrNull(it.fullName(), it.email) }
 
-        filterNewContacts(hankeId, contacts).forEach { contact -> createToken(hankeId, contact) }
+        filterNewContacts(hankeId, contacts)
+            .map { contact -> createHankeKayttaja(hankeId, contact) }
+            .also { kayttajaList ->
+                val inviter = getKayttajaByCurrentUser(hankeId)
+                sendHankeInvitationEmails(hanke, inviter, kayttajaList)
```

Comment:
Feels like sending the emails would be a logical part of `createHankeKayttaja`, since we want to send mail every time we create a new one.

Reply:
I feel that sending the email should not be part of createHankeKayttaja (createTunnisteAndKayttaja after rebase from dev). I think it is the responsibility of saveNewTokensFromHanke to control the bigger picture.

On the other hand, I do agree that it is better to send the email every time a user is created. I changed this part accordingly. 

Additionally, createTunnisteAndKayttaja would require a Hanke as parameter. That would cause changes to saveNewTokensFromApplication. Also, the current implementation does not affect saveNewTokensFromApplication as I will implement that on another pr.

Reply:
Let's see how the next PR goes.